### PR TITLE
restructured definition section, also restructured non0jquery section

### PR DESCRIPTION
**Pull request by brendajin**

### CODE REVIEW

File: ajax/index.html [hunk 1]
```diff
@@ -251,102 +252,121 @@ <h1 class="center">Asynchronous Technologies: AJAX/JSON</h1>
...
-    request = new ActiveXObject('Microsoft.XMLHTTP');
-  } else if (window["XMLHttpRequest"]) {
-    request = new XMLHttpRequest();
-  }
-} catch (e) {
-  request = false;
-}
-if (!request ) alert("Error initializing XMLHttpRequest!");
+// instantiate a new request
+var request = new XMLHttpRequest();
```
Comment (pamelafox) [inline ajax/index.html]: wooo, its happened! we no longer need to teach the old way! hallelujah!

File: ajax/index.html [hunk 2]
```diff
@@ -251,102 +252,121 @@ <h1 class="center">Asynchronous Technologies: AJAX/JSON</h1>
...
-  }
-} catch (e) {
-  request = false;
-}
-if (!request ) alert("Error initializing XMLHttpRequest!");
+// instantiate a new request
+var request = new XMLHttpRequest(); 
+
+// add event listeners
+request.onload = function() = {
```
Comment (pamelafox) [inline ajax/index.html]: hm extra = sign?

File: ajax/index.html [hunk 3]
```diff
@@ -251,102 +252,121 @@ <h1 class="center">Asynchronous Technologies: AJAX/JSON</h1>
...
+
+  // transform a string into a usable object
+  console.log( JSON.parse( request.responseText ) );
+};;
```
Comment (pamelafox) [inline ajax/index.html]: extra semi-colon?
(can you check this code parses/works)?

File: ajax/index.html (hunk 2 above)
Comment (pamelafox) [inline ajax/index.html]: what do you think about using addEventListener('load')? thats how we usually teach DOM event listeners now. does that work here?

File: ajax/index.html [hunk 4]
```diff
@@ -251,102 +252,121 @@ <h1 class="center">Asynchronous Technologies: AJAX/JSON</h1>
...
         <section class="content">
+        <p>When using POST, you will want to also send along some data.</p>
 <pre>
-function sendRequest() { 
-  url="books.json";
-  request.open("GET", url, true);
-  request.onreadystatechange = processRequest;
-  request.send(null); 
+var data = {
+  kittenNames: ['Mary-Ann', 'Brenda'],
```
Comment (pamelafox) [inline ajax/index.html]: i'm trying to figure out if this is one kitten or 2 or one kitten with 2 personalities.

File: ajax/index.html [hunk 5]
```diff
@@ -251,102 +252,121 @@ <h1 class="center">Asynchronous Technologies: AJAX/JSON</h1>
...
+var request = new XMLHttpRequest();
+
+request.onload = function() {
+  // use the .status property to check the status code
+  if ( request.status === 200 ) {
+    success( JSON.parse( request.responseText ) );
+  } else {
+    console.log( 'Error!' )
+  }
+}
```
Comment (pamelafox) [inline ajax/index.html]: missing semi-colon

File: ajax/index.html [hunk 6]
```diff
@@ -251,102 +252,121 @@ <h1 class="center">Asynchronous Technologies: AJAX/JSON</h1>
...
-    } 
-  } 
+var request = new XMLHttpRequest();
+
+request.onload = function() {
+  // use the .status property to check the status code
+  if ( request.status === 200 ) {
+    success( JSON.parse( request.responseText ) );
+  } else {
+    console.log( 'Error!' )
```
Comment (pamelafox) [inline ajax/index.html]: missing semi-colon

File: ajax/index.html [hunk 7]
```diff
@@ -251,102 +252,121 @@ <h1 class="center">Asynchronous Technologies: AJAX/JSON</h1>
...
-       request.status + ":" +
-       request.statusText); 
-    } 
-  } 
+var request = new XMLHttpRequest();
+
+request.onload = function() {
+  // use the .status property to check the status code
+  if ( request.status === 200 ) {
+    success( JSON.parse( request.responseText ) );
```
Comment (pamelafox) [inline ajax/index.html]: the JS conventions that I've had at jobs dont use the spaces inside the parentheses that you've used throughout the slides, but i understand if the point is to make the code easier to understand, it does have that effect. just want to make sure thats the reason.

File: ajax/index.html [hunk 8]
```diff
@@ -251,102 +252,121 @@ <h1 class="center">Asynchronous Technologies: AJAX/JSON</h1>
...
+  } else {
+    console.log( 'Error!' )
+  }
+}
+
+// handle errors
+request.onerror = function() {
+  console.log( 'Error!' )
 }
```
Comment (pamelafox) [inline ajax/index.html]: semi-colon

File: ajax/index.html [hunk 9]
```diff
@@ -251,102 +252,121 @@ <h1 class="center">Asynchronous Technologies: AJAX/JSON</h1>
...
-  "categories": ["scifi", "shortstories"]
-  },
-  {"title": "Pirate Cinema",
-   "author": "Cory Doctorow",
-   "categories": ["scifi", "romance"]
-  }
-]
+request.onload = function() {
+  processReponse( JSON.parse( request.responseText ) );
+}
```
Comment (pamelafox) [inline ajax/index.html]: semi-colon

File: ajax/index.html [hunk 10]
```diff
@@ -251,102 +252,121 @@ <h1 class="center">Asynchronous Technologies: AJAX/JSON</h1>
...
+          Robot Dreams, by Isaac Asimov<br>
+          Pirate Cinema, by Cory Doctorow
+        <p>
         </section>
-      </div><!--processing the response-->
+      </div><!--XMLHttpRequest response handling-->
 
       <div class="slide normal">
-        <header><h1>4) Processing the Response</h1></header>
+        <header><h1>XMLHttpRequest in IE</h1></header>
```
Comment (pamelafox) [inline ajax/index.html]: think we can remove this now, since new IE supports new way, right?

File: ajax/index.html (hunk 7 above)
Comment (brendajin) [inline ajax/index.html]: Yeah, it was more for the readability of that nested success function. I agree that it is not consistent stylistically with the rest of the slides. I'll remove the extra spacing.

File: ajax/index.html (hunk 10 above)
Comment (brendajin) [inline ajax/index.html]: Yes... I was recently asked about support for this and fallbacks... it's worth a mention on an earlier slide.

File: ajax/index.html [hunk 11]
```diff
@@ -251,94 +252,92 @@ <h1 class="center">Asynchronous Technologies: AJAX/JSON</h1>
...
-       request.status + ":" +
-       request.statusText); 
-    } 
-  } 
-}
+var kitten = {
+  name: 'Mary-Ann',
+  siblings: ['Pamela', 'Brenda', 'Bianca'],
+  id: 'ab92ed',
+  fluffy: true
```
Comment (pamelafox) [inline ajax/index.html]: lol mary ann is so fluffy, love it

**Comment by pamelafox** [comment]
LGTM, merge at will!

File: ajax/index.html (hunk 11 above)
Comment (mjawili) [inline ajax/index.html]: NooOOoOooO!  =^・ᴥ・^=

**Pull request merged**